manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorker.java

```diff
@@ -0,0 +1,175 @@
+package com.redhat.service.bridge.manager.workers.resources;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.infra.models.dto.ManagedResourceStatus;
+import com.redhat.service.bridge.manager.models.ManagedResource;
+import com.redhat.service.bridge.manager.models.Work;
+import com.redhat.service.bridge.manager.workers.WorkManagerImpl;
+import com.redhat.service.bridge.manager.workers.Worker;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+
+public abstract class AbstractWorker<T extends ManagedResource> implements Worker<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorker.class);
+
+    private static final Set<ManagedResourceStatus> PROVISIONING_COMPLETE = Set.of(ManagedResourceStatus.READY, ManagedResourceStatus.FAILED);
+    private static final Set<ManagedResourceStatus> DEPROVISIONING_COMPLETE = Set.of(ManagedResourceStatus.DELETED, ManagedResourceStatus.FAILED);
+
+    @ConfigProperty(name = "event-bridge.resources.worker.max-retries")
+    int maxRetries;
+
+    @ConfigProperty(name = "event-bridge.resources.workers.timeout-seconds")
+    int timeoutSeconds;
+
+    @Inject
+    WorkManagerImpl workManager;
+
+    @Override
+    public boolean handleWork(Work work) {
+        //By the time VertX gets to execute an item of Work it may have been deleted.
+        if (!workManager.exists(work)) {
+            return false;
+        }
+
+        T managedResource = load(work);
+        if (Objects.isNull(managedResource)) {
+            //Work has been scheduled but cannot be found. Something (horribly) wrong has happened.
+            String message = String.format("Resource of type '%s' with id '%s' no longer exists in the database.", work.getType(), work.getManagedResourceId());
+            throw new IllegalStateException(message);
+        }
+
+        boolean complete = false;
+        if (managedResource.getStatus() == ManagedResourceStatus.ACCEPTED) {
+            complete = PROVISIONING_COMPLETE.contains(createDependencies(work, managedResource).getStatus());
```

Comment:
I'd avoid inline methods in arguments that do side effects in general or in this case, that effectively do the most important method of the class

Reply:
Awwww. There goes my cryptically in-lined code.

Reply:
Updated.